zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/job/JobUpdateProcessor.java

```diff
@@ -42,26 +45,48 @@ public void processRecord(final TypedRecord<JobRecord> command) {
         .ifRightOrLeft(
             job -> {
               final List<String> errors = new ArrayList<>();
-              jobUpdateBehaviour.updateJobRetries(jobKey, job, command).ifPresent(errors::add);
-              final long timeout = command.getValue().getTimeout();
-              // if no timeout is provided (the default value is -1L), no update
-              if (timeout > 0) {
-                jobUpdateBehaviour.updateJobTimeout(jobKey, job, command).ifPresent(errors::add);
-              }
+              final Map<String, Number> changeset = command.getValue().getChangedAttributes();
+              jobChange(
+                  changeset,
+                  JobRecord.RETRIES,
+                  Number::intValue,
+                  (retries) -> jobUpdateBehaviour.updateJobRetries(jobKey, retries, job, command),
+                  errors);
+              jobChange(
+                  changeset,
+                  JobRecord.TIMEOUT,
+                  Number::longValue,
+                  (timeout) -> jobUpdateBehaviour.updateJobTimeout(jobKey, timeout, job, command),
+                  errors);
               if (errors.isEmpty()) {
                 stateWriter.appendFollowUpEvent(jobKey, JobIntent.UPDATED, job);
                 responseWriter.writeEventOnCommand(jobKey, JobIntent.UPDATED, job, command);
               } else {
-                final String errorMessage = String.join(", ", errors);
-                rejectionWriter.appendRejection(
-                    command, RejectionType.INVALID_ARGUMENT, errorMessage);
-                responseWriter.writeRejectionOnCommand(
-                    command, RejectionType.INVALID_ARGUMENT, errorMessage);
+                handleRejection(errors, command);
               }
             },
             errorMessage -> {
               responseWriter.writeRejectionOnCommand(
                   command, RejectionType.NOT_FOUND, errorMessage);
             });
   }
+
+  private <T extends Number> void jobChange(
+      final Map<String, Number> changeset,
+      final String key,
+      final Function<Number, T> converter,
+      final Function<T, Optional<String>> updateFunction,
+      final List<String> errors) {
+    if (changeset.containsKey(key)) {
+      final T value = converter.apply(changeset.get(key));
```

Comment:
🔧 Should we handle any conversion issues that might occur? We should also test what happens when the conversion does not work properly.